c10/util/BFloat16-math.h

```diff
@@ -91,4 +91,56 @@ inline c10::BFloat16 fmod(c10::BFloat16 a, c10::BFloat16 b) {
   return std::fmod(float(a), float(b));
 }
 
+C10_HOST_DEVICE inline c10::BFloat16 nextafter(
+    c10::BFloat16 from,
+    c10::BFloat16 to) {
+  using int_repr_t = uint16_t;
+  using float_t = c10::BFloat16;
+  constexpr uint8_t bits = 16;
+  union {
+    float_t f;
+    int_repr_t i;
+  } ufrom = {from}, uto = {to};
+
+  // get a mask to get the sign bit i.e. MSB
+  int_repr_t sign_mask = int_repr_t{1} << (bits - 1);
+
+#if defined(__HIP_PLATFORM_HCC__) || defined(_MSC_VER)
```

Comment:
`#ifdef` roughly copied from
https://github.com/pytorch/pytorch/blob/f91288972682b5f7157f40df2d7ac7536d456124/c10/util/BFloat16.h#L52-L58

However, I am not sure but was unable to use `isnan` on MSVC.
Build failure (calling host function from host device function) : https://github.com/pytorch/pytorch/runs/3106706944

Reply:
Why bother with isnan instead of comparing for equality?

Reply:
You mean consistently use `x != x` to verify whether `x` is NaN or not? (if yes, then it is not very clear that we are testing for NaN)

Reply:
I think a reasonable reader will understand `x != x` as a test for NaN. And yep, that is what I meant